src/hubverse_transform/model_output.py

```diff
@@ -236,31 +260,43 @@ def parse_file(cls, file_name: str) -> dict:
 
     def read_file(self) -> pa.table:
         """Read model-output file into PyArrow table."""
-
         logger.info(f"Reading file: {self.input_file}")
-
         if self.file_type == ".csv":
             model_output_file = self.fs_input.open_input_stream(self.input_file)
+            schema = self._get_schema(self.tasks, self.file_type, model_output_file)
             # normalize incoming missing data values to null, regardless of data type
             options = csv.ConvertOptions(
                 null_values=["na", "NA", "", " ", "null", "Null", "NaN", "nan"],
                 strings_can_be_null=True,
-                # temp fix: force location and output_type_id columns to string
-                column_types={"location": pa.string(), "output_type_id": pa.string()},
-            )
+                column_types=schema)
             model_output_table = csv.read_csv(model_output_file, convert_options=options)
         else:
-            # temp fix: force location and output_type_id columns to string
             model_output_file = self.fs_input.open_input_file(self.input_file)
-            schema_new = pq.read_schema(model_output_file)
-            for field_name in ["location", "output_type_id"]:
-                field_idx = schema_new.get_field_index(field_name)
-                if field_idx >= 0:
-                    schema_new = schema_new.set(field_idx, pa.field(field_name, pa.string()))
-            model_output_table = pq.read_table(model_output_file, schema=schema_new)
+            schema = self._get_schema(self.tasks, self.file_type, model_output_file)
+            model_output_table = pq.read_table(model_output_file, schema=schema)
 
         return model_output_table
 
+
+    @classmethod
+    def _get_schema(cls, tasks, file_type, model_output_file):
+        """
+        read_file() helper that returns a pa.schema. uses my tasks if they were able to be loaded. o/w we hard-code a
+        few common hub columns and take our chances!
+        """
+        if tasks:
+            schema = create_hub_schema(tasks)
+        elif file_type == ".csv":
```

Comment:
Personally I don't think this is necessary for a couple of reasons:
1.  It is safe to assume that any hub supporting a sync cloud has (and should have) valid config. This helps ensure files are guaranteed to be copied over correctly, regardless of what custom task IDs they may be collecting
2. The particular fix has always been temporary and hacky. It cannot for example ensure problems don't arise in custom task IDs. Only knowledge of the schema would ensure that. As such I don't feel it should be part of our official transform function any more.

As such, I propose throwing an error if a task.json cannot be found. Otherwise we continue to remain open to issues in custom task IDs that we may not find out about until someone (likely a user) tries to filter on such a column.

Reply:
Agreed!

Reply:
Agreed too!